File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.java
class Solution {
    public long solution(int n) {
        
        long answer = 0;
        long[] dp = new long[n+2];
        
        dp[1] = 1;
        dp[2] = 2;
        
        if (n == 1) {answer = 1;}
        if (n == 2) {answer = 2;}
        
        for(int i = 3; i <= n; i++) {
        dp[i] = (dp[i-1] + dp[i-2])%1234567;
        }
    
        answer = dp[n];

        return answer;
    }
}

File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.js
function solution(n) {
    var answer = 0;
   // 1, 2 합해서 4가 나올수 있는 경우
    let dp = [1,1,2];
    
    for(let i = 3 ; i <=n ;i ++){
        dp[i] =(dp[i-1] + dp[i-2])% 1234567;
    }
    
    //1234567나누기;
    return dp[n];
}